hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -584,7 +589,8 @@ def onchange_unit_amount(self):
             new_ts.write({'unit_amount': unit_amount})
         else:
             new_ts_values = self._line_to_timesheet(diff_amount)
-            self.env['account.analytic.line'].create(new_ts_values)
+            if new_ts_values:
```

Comment:
because now `self._line_to_timesheet` can return False in `hr_timesheet_sheet_no_negative`.

Reply:
You could return `{}` instead, so you can avoid this `if`.

Reply:
No, because then `self.env['account.analytic.line'].create({})` would create a timesheet. Well, in fact, it will gives an error because no required parameter passed in the create.